### PR TITLE
Add GitHub Action for continuous integration testing

### PR DESCRIPTION
**Pull request by Corvince**
Given the issue of travis.org shutting down on Dec. 31st 2020, which is just 2 weeks away (tracked in #938), this PR would add the github actions required to continue our testing as usual with GA. 

I know #938 is not yet resolved and I don't take any damage if we decide not to use GA, but CircleCI or something else (but the current vote favors GA so I went ahead).

### CODE REVIEW

**Comment by jackiekazil** [comment]
Thank you @Corvince! I was just digging through issues trying to catch up. 
What do you think is the best way to "test" this? Other than.... 

1. Is it working with GA? 
2. Does it do at least what the old stuff did?

**Comment by rht** [comment]
Is GA not yet enabled? I can't find the GA logs. The coverage was probably generated by Travis. Maybe also delete .travis.yml in this PR?

**Comment by Corvince** [comment]
Good idea to delete the travis configuration.

The GA are working the logs are here (for my fork)
https://github.com/Corvince/mesa/actions/runs/454126773

Currently it does what we did on travis: runs flake8, black and pytest and submits the code coverage to coveralls.

Runs on 3.6--3.9 on ubuntu

**Comment by rht** [comment]
Tests look good, e.g. https://github.com/Corvince/mesa/runs/1629438627?check_suite_focus=true#step:7:39.

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install codecov black
+        pip install .[dev]
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (rht) [inline .github/workflows/python.yml:24]: Why is this `if` clause required?

Comment (Corvince) [reply .github/workflows/python.yml:24]: Ah, it's an artifact of the GA template. Don't know why I kept it, mesa doesn't have any requirements file. Which also explains why there is an if statement, but not why we need it. Should I delete the line?

Comment (rht) [reply .github/workflows/python.yml:24]: IMO just delete it. It will become confusing in the future without proper comment explaining why.

Comment (Corvince) [reply .github/workflows/python.yml]: You are probably right, I deleted it

**Comment by jackiekazil** [comment]
This looks good to me. 
@Corvince, i am going to merge to push forward. Can you submit another PR with the line deletion that @rht caught?

**Pull request merged**